### PR TITLE
Fix: Reset the staking data when switching wallets

### PR DESCRIPTION
**Pull request by LeifuChen**
## Description
1. The staking data will be reset when disconnecting the wallets
2. The staking data will be refreshed immediately when switching the wallets
3. The eligibility of trading rewards should be verified by staking v1 data 
4. The claimable rewards will be fetched when switching the wallets regardless of the pages
5. Remove the redemption-related components
6. Remove the reverse migration banner

## Related issue
#2694

### CODE REVIEW

**Review by platschi — APPROVED** [review]
Works great for both disconnecting wallet reset and refresh when switching wallets with staking balances.

**Review by Marko-roy — APPROVED** [review]

File: packages/app/src/state/staking/actions.ts [hunk 1]
```diff
@@ -8,103 +6,92 @@ import { monitorTransaction } from 'contexts/RelayerContext'
...
+	EstimatedRewards,
+	StakingAction,
+	StakingActionV2,
+} from './types'
+
+export const fetchStakingData = createAsyncThunk<StakingAction, void, ThunkConfig>(
+	'staking/fetchStakingData',
+	async (_, { getState, extra: { sdk } }) => {
+		try {
+			const { wallet } = getState()
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply packages/app/src/state/staking/actions.ts]: True. Updated.

File: packages/app/src/state/staking/actions.ts [hunk 2]
```diff
@@ -8,103 +6,92 @@ import { monitorTransaction } from 'contexts/RelayerContext'
...
-			stakedEscrowedBalance: stakedEscrowedBalance.toString(),
-			claimableBalance: claimableBalance.toString(),
-			totalStakedBalance: totalStakedBalance.toString(),
-			stakedResetTime,
-			kwentaStakingV2Allowance: kwentaStakingV2Allowance.toString(),
+export const fetchStakingV2Data = createAsyncThunk<StakingActionV2, void, ThunkConfig>(
+	'staking/fetchStakingDataV2',
+	async (_, { getState, extra: { sdk } }) => {
+		try {
+			const { wallet } = getState()
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply packages/app/src/state/staking/actions.ts]: Updated.

File: packages/app/src/state/staking/actions.ts [hunk 3]
```diff
@@ -142,82 +129,88 @@ export const redeemToken = createAsyncThunk<void, 'vKwenta' | 'veKwenta', ThunkC
...
-				vestable: e.vestable.toString(),
-				amount: e.amount.toString(),
-				fee: e.fee.toString(),
-			})),
-			totalVestable: totalVestable.toString(),
+export const fetchEscrowData = createAsyncThunk<EscrowBalance, void, ThunkConfig>(
+	'staking/fetchEscrowData',
+	async (_, { getState, extra: { sdk } }) => {
+		try {
+			const { wallet } = getState()
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply packages/app/src/state/staking/actions.ts]: Updated.

File: packages/app/src/state/staking/actions.ts [hunk 4]
```diff
@@ -142,82 +129,88 @@ export const redeemToken = createAsyncThunk<void, 'vKwenta' | 'veKwenta', ThunkC
...
-				vestable: e.vestable.toString(),
-				amount: e.amount.toString(),
-				fee: e.fee.toString(),
-			})),
-			totalVestable: totalVestable.toString(),
+export const fetchEscrowV2Data = createAsyncThunk<EscrowBalance, void, ThunkConfig>(
+	'staking/fetchEscrowV2Data',
+	async (_, { getState, extra: { sdk } }) => {
+		try {
+			const { wallet } = getState()
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply packages/app/src/state/staking/actions.ts]: Updated.

File: packages/app/src/state/staking/actions.ts [hunk 5]
```diff
@@ -315,52 +308,41 @@ export const compoundRewards = createAsyncThunk<void, void, ThunkConfig>(
...
-			kwentaRewards: kwentaRewards.toString(),
-			opRewards: opRewards.toString(),
-			snxOpRewards: snxOpRewards.toString(),
+export const fetchClaimableRewards = createAsyncThunk<ClaimableRewards, void, ThunkConfig>(
+	'staking/fetchClaimableRewards',
+	async (_, { getState, extra: { sdk } }) => {
+		try {
+			const {
+				staking: { epochPeriod },
+				wallet,
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply packages/app/src/state/staking/actions.ts]: Updated.

File: packages/app/src/state/staking/types.ts [hunk 6]
```diff
@@ -2,50 +2,70 @@ import { EscrowData, ClaimParams } from '@kwenta/sdk/types'
...
 	totalStakedBalance: string
+	claimableBalance: string
 	stakedEscrowedKwentaBalance: string
-	stakedKwentaBalance: string
-	totalVestable: string
-	escrowData: EscrowData<string>[]
 }
 
-export type StakingState = {
+type StakingTransaction = {
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply packages/app/src/state/staking/types.ts]: No. They are just the miscellaneous info about the staking status. Updated.

**Review by platschi — APPROVED** [review]
Works good here 🙏🏼

**Review by avclarke — APPROVED** [review]

**Pull request merged**